src/service/ticket.ts

```diff
@@ -0,0 +1,53 @@
+import { FastifyReply, FastifyRequest } from "fastify";
+import { Ticket } from "../entities";
+
+/**
+ * Retrieves a ticket from DB. It returns error either;
+ * 1. Passed ticket id is invalid - not convertible to Number
+ * 2. Ticket couldn't be found by the passed id
+ */
+export async function getTicketById(ticketId: string): Promise<Ticket> {
+  const ticketIdQuery = Number(ticketId);
+
+  if (isNaN(ticketIdQuery) || !ticketIdQuery) throw new Error(`Invalid ticket id: ${ticketIdQuery}`);
```

Comment:
you throw an error with `Invalid ticket id: NaN` if the isNaN(ticketIdQuery) is true. Is it your intention?

or you want to throw an error with `Invalid ticket id: ${ticketId}`

Reply:
Yeah that's an intention. Not sure what's the best practice around here, but I'd figure that's helpful if it says NaN instead of regular string error message since that indicates request is not parsible to number. It's _tiny a bit_ more informative than regular text error message. I can change it though.

Reply:
@ryonryon I'll merge it but lemme know if you think there's any better way, I'd love to hear & fix!